ambry-frontend/src/main/java/com.github.ambry.frontend/AmbrySecurityService.java

```diff
@@ -232,7 +230,17 @@ private void setGetBlobResponseHeaders(BlobInfo blobInfo, GetBlobOptions options
         restResponseChannel.setHeader("Content-Disposition", "attachment");
       }
     }
-    if (blobProperties.isPrivate()) {
+    setCacheHeaders(blobProperties.isPrivate(), restResponseChannel);
+  }
+
+  /**
+   * Sets headers that provide directions to proxies and caches.
+   * @param isPrivate whether the blob is private.
+   * @param restResponseChannel the channel that the response will be sent over
+   * @throws RestServiceException if there is any problem setting the headers
+   */
+  private void setCacheHeaders(boolean isPrivate, RestResponseChannel restResponseChannel) throws RestServiceException {
```

Comment:
I guess we would never send a not-modified response for blobInfo or userMetadata